pkg/controller/replicaset/replica_set.go

```diff
@@ -567,26 +567,17 @@ func (rsc *ReplicaSetController) syncReplicaSet(key string) error {
 	}()
 
 	obj, exists, err := rsc.rsLister.Indexer.GetByKey(key)
+	if err != nil {
```

Comment:
@deads2k doesn't this change mean that we are treating all errors here as notFound? Should we cherry-pick this in 1.5?